sci-mathematics/msieve/msieve-1.53.recipe

```diff
@@ -0,0 +1,63 @@
+SUMMARY="Library for Factoring Large Integers"
+DESCRIPTION="C library implementing a suite of algorithms to factor large integers"
+HOMEPAGE="https://sourceforge.net/projects/msieve/"
+COPYRIGHT="Public Domain"
+LICENSE="Public Domain"
+REVISION="1"
+SOURCE_URI="https://downloads.sourceforge.net/msieve/msieve/Msieve%20v$portVersion/msieve${portVersion/./}_src.tar.gz"
+CHECKSUM_SHA256="c5fcbaaff266a43aa8bca55239d5b087d3e3f138d1a95d75b776c04ce4d93bb4"
+PATCHES="msieve-1.53.patchset"
+
+ARCHITECTURES="?x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	msieve$secondaryArchSuffix = $portVersion
+	cmd:msieve$secondaryArchSuffix
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libecm$secondaryArchSuffix
+	lib:libgmp$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	msieve${secondaryArchSuffix}_devel = $portVersion
+	devel:libmsieve$secondaryArchSuffix = $portVersion
+	"
+REQUIRES_devel="
+	haiku${secondaryArchSuffix}_devel
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libecm$secondaryArchSuffix
+	devel:libgmp$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:make
+	cmd:gcc$secondaryArchSuffix
+	cmd:sed
+	"
+
+PATCH()
+{
+	# Allow owns CFLAGS, could be useful when doing debug builds
+	sed -e 's/^CFLAGS =/CFLAGS +=/' -i Makefile
+}
+
+BUILD()
+{
+	make $jobArgs all ECM=1 HAIKU=1
+}
+
+INSTALL()
+{
+	install -Dm755 msieve "$binDir"/msieve
+	mkdir -p "$includeDir"
+	cp include/* "$includeDir"
+	install -Dm755 libmsieve.a "$developLibDir/libmsieve.a"
```

Comment:
What do you think of the following variant?
```sh
INSTALL()
{
	install -d -m 755 "$binDir" "$docDir" "$includeDir" "$developLibDir"
	install -t "$binDir" -m 755 msieve
	install -t "$docDir" -m 644 Changes Readme*
	install -t "$includeDir" -m 644 include/* 
	install -t "$developLibDir" -m 644 libmsieve.a
	packageEntries devel "$developDir"
}
```
Yep, for `libmsieve.a` I guess you meant 644, not 755.

BTW, in PROVIDES_DEVEL, maybe we should remove the "` = $portVersion`" in:
```
	devel:libmsieve$secondaryArchSuffix = $portVersion
```

Hmm, if you feel like trying to build libmsieve as a shared lib, that could be nice.
I can try to help you with that, but I'm sure this would be a piece of cake for some mentors.

Reply:
It could be done, but virtually no distribution does this, and not having a shared lib might means that the ABI/API can break at anytime, so we shouldn't do this.